### PR TITLE
move hasNonConstPointers() from init.d to initsem.d

### PR DESCRIPTION
**Pull request by WalterBright**
Because it is only ever called by initsem.d and so belongs there. It makes no sense for it to be in init.d.

A pure cut&paste job, except added missing Ddoc comment and made it `private` and removed the unnecessary `extern (C++)`.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: src/dmd/init.d [hunk 1]
```diff
@@ -217,76 +217,6 @@ extern (C++) final class ExpInitializer : Initializer
     }
 }
 
-version (all)
-{
-    extern (C++) bool hasNonConstPointers(Expression e)
```
Comment (thewilsonator) [inline src/dmd/init.d:222]: this may need to be removed from the headers

**Review by thewilsonator — APPROVED** [review]
Otherwise looks good.

File: src/dmd/init.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/init.d:222]: ok

**Comment by ibuclaw** [comment]
Don't think this is used outside of the frontend.

**Pull request merged**